test/lib/recordevents/observer/observer.go

```diff
@@ -132,15 +138,15 @@ func (o *Observer) ServeHTTP(writer http.ResponseWriter, request *http.Request)
 	eventInfo := recordevents.EventInfo{
 		Error:       eventErrStr,
 		Event:       event,
-		HTTPHeaders: header,
+		HTTPHeaders: headers,
 		Origin:      request.RemoteAddr,
 		Observer:    o.Name,
 		Time:        time.Now(),
 		Sequence:    atomic.AddUint64(&o.seq, 1),
 	}
 	err := o.EventLogs.Vent(eventInfo)
 	if err != nil {
-		logging.FromContext(o.ctx).Warnw("Error while venting the recorded event", zap.Error(err))
+		logging.FromContext(o.ctx).Fatalf("Error while venting the recorded event", zap.Error(err))
```

Comment:
nit, I think this is supposed to be Fatalw since you're using zap.* or just Fatal? Surprised the linter didn't complain though, been seeing those before.